exercises/dominoes/.meta/src/reference/java/Dominoes.java

```diff
@@ -0,0 +1,69 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+
+class Dominoes {
+	
+	ArrayList<Domino> formChain(ArrayList<Domino> inputDominoes) {
+		if (inputDominoes.size() == 0) {
+			throw new IllegalArgumentException("Empty input domino list.");
+		} else {
+			ArrayList<Domino> dominoChain = formPartialChain(new ArrayList<Domino>(), inputDominoes);
+			if(dominoChain == null) {
+				throw new IllegalArgumentException("No domino chain found.");
```

Comment:
Please add a whitespace after `if` 🙂 

Also I'm not sure if `IllegalArgumentException` is the correct exception to throw here. It's not really an illegal argument, it's just that it's not possible to form a chain with these dominoes. Maybe it would be better to use  a custom exception? What do you think @jssander, @exercism/java?

Reply:
I'd agree that a custom exception would be more sensible! It's a completely valid argument I think - it's more a dominoes-specific rule, right? :slightly_smiling_face: